source/integrations/mongoose-get-started.txt

```diff
@@ -0,0 +1,642 @@
+      :language: javascript
+      :start-after: start-populate-author
+      :end-before: end-populate-author
+
+   .. output::
+      :language: console
+      :visible: false
+
+      Article Populated: {
+        _id: new ObjectId('...'),
+        title: 'Awesome Post!',
+        slug: 'awesome-post',
+        published: false,
+        author: {
+          _id: new ObjectId('...'),
+          name: 'Jess Garica',
+          email: 'jgarcia@email.com',
+          __v: 0
+        },
+        content: 'This is the best post ever',
+        tags: [ 'featured', 'announcement' ],
+        createdAt: 2025-05-15T18:04:28.590Z,
+        comments: [],
+        __v: 0
+      }
+
+This populates the data for the ``author`` field in the data for this article.
+Mongoose uses the MongoDB ``$lookup`` method to do this under the hood.
+
+Middleware
+----------
+
+In Mongoose, middleware are functions that run before, and, or during the
+execution of asynchronous functions at the schema level.
+
+To use middleware in this project, go to the ``Blog.js`` file. Here you can set
+the ``updated`` date field to update every time an article is saved or updated
+by using middleware.
+
+Add the following code to a line below your ``blogSchema`` declaration:
+
+.. literalinclude:: /includes/integrations/mongoose-blogSchema-validate.js
+   :language: javascript
+   :start-after: start-blogSchema-middleware
+   :end-before: end-blogSchema-middleware
+   :dedent:
+
+Go to the ``index.js`` file and then add the following code to find an article,
+update the title, and then save the updated article:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/integrations/mongoose-get-started-multiple-schemas.js
+      :language: javascript
+      :start-after: start-middleware-update
+      :end-before: end-middleware-update
+
+   .. output::
+      :language: console
+      :visible: false
+
+      Article Updated: {
+        _id: new ObjectId('...'),
+        title: 'Updated Title',
+        slug: 'awesome-post',
+        published: false,
+        author: new ObjectId('...'),
+        content: 'This is the best post ever',
+        tags: [ 'featured', 'announcement' ],
+        createdAt: 2025-05-15T18:04:28.590Z,
+        comments: [],
+        __v: 0,
+        updated: 2025-05-15T18:18:30.249Z
+      }
+
+When you run the application, the returned article includes an ``updated`` date.
+
+Besides the ``pre()`` middleware function, Mongoose also offers a ``post()``
+middleware function.
+
+Resources and Next Steps
+------------------------
+
+You now have a sample project that uses Mongoose to perform CRUD operations on a
+MongoDB collection. From here, you can choose to build on this project to
+incorporate :ref:`Mongoose helper methods
+<node-mongoose-get-started-helper-methods>` and build more complex queries.
+
+To learn more about using Mongoose with MongoDB, see the `Mongoose documentation
+<https://mongoosejs.com/docs/guide.html>`__.
```

Comment:
If not converting this into a stepped tutorial, I would suggest converting this to an "Additional Information" section